config/routes.rb

```diff
@@ -23,11 +25,15 @@
     # Newer implementation: will supersede the above.
     get "users/two_factor_authentication/setup", to: "users/two_factor_authentication/setup#show"
     put "users/two_factor_authentication/setup", to: "users/two_factor_authentication/setup#update"
-    get "users/two_factor_authentication/email", to: "users/two_factor_authentication/email#show"
-    post "users/two_factor_authentication/email", to: "users/two_factor_authentication/email#update"
+
+    get "users/two_factor_authentication/email/resend", to: "users/two_factor_authentication/email#resend"
+    get "users/two_factor_authentication/email/confirm", to: "users/two_factor_authentication/email#confirm"
+    get "users/two_factor_authentication/app/totp_with_qr", to: "users/two_factor_authentication/app#totp_with_qr"
```

Comment:
Following the same reasoning as above, this path should ideally be nested within `/setup` as it's part of the setup process (e.g. `/setup/app`). 

Why is `totp_with_qr` needed? Can we not do without it? I'm not entirely clear.

Reply:
totp_with_qr is needed because it displays the QR of the secret key, which should only be shown at setup.

Reply:
renamed it to the default show